crates/sui-framework/src/lib.rs

```diff
@@ -12,6 +12,8 @@ pub mod natives;
 
 static SUI_FRAMEWORK_BYTES: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/sui-framework"));
 static SUI_FRAMEWORK: Lazy<Vec<CompiledModule>> = Lazy::new(|| {
+    // TODO: deserialization uses VERSION_MAX of supported Move bytecode version rather than the max
```

Comment:
Yep, I believe these codepaths are fine to keep as-is.

Reply:
Where are these modules used?

Reply:
They are used whenever the validator needs to load the framework from its own binary, i.e. genesis and framework upgrades.  So, my rationale is that in both those scenarios we should be doing the check when we access the module, rather than when we create it.